runtime/lua/vim/_extui/messages.lua

```diff
@@ -405,17 +405,24 @@ function M.set_pos(type)
       local row = (texth.all > height and texth.end_row or 0) + 1
       api.nvim_win_set_cursor(ext.wins[ext.tab].box, { row, 0 })
     elseif type == 'more' and api.nvim_get_current_win() ~= win then
-      api.nvim_create_autocmd('WinEnter', {
-        once = true,
-        callback = function()
-          if api.nvim_win_is_valid(win) then
-            api.nvim_win_set_config(win, { hide = true })
-          end
-          append_more = 0
-        end,
-        desc = 'Hide inactive more window.',
-      })
-      api.nvim_set_current_win(win)
+      -- Cannot leave the cmdwin to enter the "more" window, so close it.
+      if fn.getcmdwintype() ~= '' then
+        api.nvim_win_close(0, true)
```

Comment:
A _slightly_ pedantic nitpick (sorry! :grin:), but it's possible for the cmdwin to not be curwin (but only temporarily).

E.g: (run this one manually)

```vim
lua require'vim._extui'.enable{}
let s:w = win_getid()
call feedkeys('q:', 'nt')
au CmdwinEnter * ++once call win_execute(s:w, 'messages')
```

_Should_ be able to get around that by just using a plain `:quit` instead (it prioritizes the cmdwin regardless of curwin), but looping through the list of windows to `nvim_win_close` the one with `win_gettype() == "command"` may be more robust.

Yeah, the cmdwin is a bit jank...